added documentation for PyDP naive bayes comparison

dnabanita7 opened the pull request:
## Description
added docs for PyDP comparison

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

chinmayshah99 committed:
upgraded sha; added delta

dnabanita7 committed:
added documentation for PyDP naive bayes comparison

gilangrilhami committed:
Fix Windows PyPi build GitHub Action (#356)

* change setup msbuild to version 1.0.2

* Test remove python 3.5 in Windows workflow

* Add back python 3.5 for Windows build

chinmayshah99 committed:
added delta support

chinmayshah99 committed:
removed deprecated mathods

chinmayshah99 committed:
fixed binding for distribution

chinmayshah99 committed:
Fixed error reporting inline with latest abseil

chinmayshah99 committed:
added delta as input in tests

chinmayshah99 committed:
added delta

chinmayshah99 committed:
Merge pull request #361 from chinmayshah99/sha-upgrade-may-1

Upgraded Google DP SHA

sid-alluri committed:
fixed mypy issues (#362)

* fixed mypy issues

* fixed mypy issues with fixed style

grilhami commented:
@dnabanita7 Great work on adding those descriptions! However, I would like to suggest some minor adjustments.

1.  I think it would be a good idea to mention the description of some variables in a code block like `this`. For example, instead of just "clf", we can do `clf`. I think that would help make the description more readable.

2.  For the part where `mean_train_accs`,  `mean_test_accs`, and `dp_results`, it'll be great if we can specify that those results are differentially private since the `GaussianNB` class from PyDP is differentially private.

dnabanita7 committed:
added inline documentation for naive_bayes.py (#360)

* added inline documentation for naive_bayes.py

* corrected parse errors

madhavajay committed:
Fixing CI (#363)

* Touch

* Touch back

* Updated GitHub CI

- Ran black black-nb mypy
- Removed python 3.5
- Fixed issue with MacOS Build using --features=-supports_dynamic_linker
- Updated Google DP to the latest commit
- Renamed Bazel target from bindings_test to pydp
- Updated pyproject.toml
- Fixed com_google_protobuf issue in WORKSPACE
- Disabled docker build on every commit / push

* Added pre-commit yaml file

- Ran pre-commit to fix lots of code issues
- Removed unused # type: ignore
- Updated setup.cfg with isort, mypy and flake8 rules
- Ran isort
- Disabled notebook tests
- Fixed matrix.os template variable in tests.yml

* Update docker to use python 3.9

- Fix mypy issues with missing typeshed
- Fixing tests.yml

* Fixing Dockerfile path to bazelisk

- Fixing issue with converting matrix.os to bazel config name

* Add ubuntu to .bazelrc file

* Fixing matrix.os conversion to bazel config

* Fixing multi-os issues in tests.yml

* More multi-os fixes

* Fixing windows bazel

* Fix windows

* Adding bazel caching for linux

- Fixing install wheel on windows
- Fixing print version

* More CI fixes

* tunshell

* Fixing macos wheel

* remove tunshell

* Quickfix serialized test data

* Added bazel-cache

- Combined publish actions into a single github action
- Added nightly tests for older versions of python
- Updated some package meta-data
- Bumped version to 1.0.3
- Added poetry.lock file to speed up CI

* Removed unused exclusion rule from pre-commit

* Adding other OSes to the Bazel Cache step

* Update test to include information about cached result from
tests/algorithms/test_bounded_mean/test_bounded_mean_int64_data.proto

* Refactored Docker and Docs actions

- Updated poetry lock file

* Renaming a few action names

- Testing versions cron

* tunshell into windows machine

* tunshell into mac

* Revert cron job for versions

- Remove tunshell for mac

* Trying specific version of cache plugin for windows tar cache error

* Added proper bazel cache directories for each OS plus Windows fix

* Trying different tar.exe fix for Windows

* Forgot to remove --output_base ./bazel-cache

* Trying windows tar fix

- Busting cache on linux and mac

* Trying older bazel cache strategy

* Debugging Windows Bazel Cache

* Trying actions/cache@v2.1.5

* Trying a different windows bazel cache approach

* Fix test yaml

- Trying Windows Bash tunshell

* woops checkout not cache

* Wrong runner

* Windows good times

* Windows cache might be working

* Try without symlinks?

* Putting cache code in all workflows

- Time to test Windows cache retrieval

* Added linting requirement back to main tests

* Fixed PR change requests

* Replace bazel with bazelisk

- Use dp submodule SHA

* debugging versions check

* Adding python binary and site-packages for different versions on unix

* Test all versions on mac and linux

* Adding Windows support for python path detection

* Reversed quotes and removed \ line endings

* Changing the var name on windows

* Using old syntax for the PYTHONPATH on windows

* Re-enabled windows on versions

* Replacing incorrect Capital R in replace

Co-authored-by: Lukasz Kozuchowski <lukasz.kozuchowski@gmail.com>

Commit:
Merge branch 'dev' of github.com:dnabanita7/PyDP into dev
merge conflicts.

Commit:
made requested changes

dnabanita7 commented:
cc @chinmayshah99 @grilhami

Commit:
broken links fix
